### ISSUE TITLE
SQL join on different data types should be ill-defined

### ISSUE THREAD

**Issue by SimonBin** [kg-construct/rml-core#124]
Currently the case for example in https://github.com/kg-construct/rml-core/blob/dd2c79428b5ae1702a3f1f4104b7b7977db32728/test-cases/RMLTC0009a-MySQL/resource.sql

```sql
CREATE TABLE student (
  Sport VARCHAR(50),
);
CREATE TABLE sport (
  ID INTEGER,
);
```

```ttl
joinCondition: [ rml:parent "ID"; rml:child "Sport" ]
```

Already fixed in PostgreSQL #115  

IMO it should be illegal in RML even if MySQL happens to so accept it

raised by @aklakan in the chat

**Comment by SimonBin** [comment]
nb. the source according to @pmaria https://www.w3.org/2001/sw/rdb2rdf/test-cases/#D009-2tables1primarykey1foreignkey has the correct types

**Comment by DylanVanAssche** [comment]
Thanks for the issue!

I also agree that ill-formed joins must be rejected. We need to define such 'ill-formed joins' as they can also happen for example in XML or JSON. What happens then if you have a string `"5"` and integer `5`?

RML-Core should define this as joins are part of RML-Core regarding the ReferenceObjectMap and also RML-Logical-View.

CC: @pmaria @andimou @elsdvlee

**Comment by dachafra** [comment]
I think this issue should be fixed in #78, what do you think @elsdvlee @pmaria?

**Comment by pmaria** [comment]
> I think this issue should be fixed in #78, what do you think @elsdvlee @pmaria?

@dachafra yes we should be able to incorporate it in that PR eventually. First we need to do some more fundamental (re)writing.

**Comment by dachafra** [comment]
This depends on the reference formulation of each data format. We move this into rml-io-registry to remind us that the RFs have to clarify this aspect.
